solidjs-tailwind/src/pages/Signin.jsx

```diff
@@ -0,0 +1,40 @@
+import { createEffect, createResource } from 'solid-js';
+import { useAuth } from '../auth';
+import { useNavigate } from '@solidjs/router';
+
+const fetchToken = () =>
+  fetch('http://localhost:4000/api/auth/token', {
+    credentials: 'include',
+  })
+    .then((response) => response.json())
+    .then((data) => data.access_token);
+
+const SigninPage = () => {
+  const { setAuth } = useAuth();
+  const navigate = useNavigate();
+  const [token] = createResource(fetchToken);
+  createEffect(() => {
+    if (!token()) {
+      return;
+    }
+
+    setAuth({ token: token() });
+    navigate(sessionStorage.getItem('auth_return_path'));
+  });
+
+  return (
+    <main class="absolute bg-black text-gray-500 w-full h-full m-0 p-0 table">
+      <form class="p-64 block mx-auto mb-2">
+        <a
+          href="http://localhost:4000/api/auth/signin?redirect_url=http%3A%2F%2Flocalhost%3A3000"
```

Comment:
same, env variable

Reply:
how did we get this? `http%3A%2F%2Flocalhost%3A3000`